src/js/player.js

```diff
@@ -2679,57 +2679,138 @@ class Player extends Component {
 
-      // Trigger the ControlBar equivalent for each key, if possible
-      } else if (keycode.isEventKey(event, 'f')) {
-        // "f" = toggle fullscreen
+      } else {
 
-        event.preventDefault();
+        this.handleHotkeys(event);
 
-        const cb = this.player_.controlBar;
-        const fullscreenToggle = cb && cb.fullscreenToggle;
+      }
+    }
+  }
 
-        if (fullscreenToggle && fullscreenToggle.handleClick) {
-          fullscreenToggle.handleClick();
-        }
-      } else if (keycode.isEventKey(event, 'm')) {
-        // "m" = toggle mute
+  /**
+   * Called when this Player receives a hotkey keydown event.
+   * Supported player-wide hotkeys are:
+   *
+   *   f          - toggle fullscreen
+   *   m          - toggle mute
+   *   k or Space - toggle play/pause
+   *
+   * @param {EventTarget~Event} event
+   *        The `keydown` event that caused this function to be called.
+   */
+  handleHotkeys(event) {
+    const hotkeys = this.options_.userActions ? this.options_.userActions.hotkeys : null;
+    let fullscreenKey;
+    let muteKey;
+    let playPauseKey;
 
-        event.preventDefault();
+    if (hotkeys) {
+      fullscreenKey = hotkeys.fullscreenKey;
+      muteKey = hotkeys.muteKey;
+      playPauseKey = hotkeys.playPauseKey;
+    }
 
-        const cb = this.player_.controlBar;
-        const volumePanel = cb && cb.volumePanel;
-        const muteToggle = (cb && cb.muteToggle) || (volumePanel && volumePanel.muteToggle);
+    if (typeof fullscreenKey !== 'function') {
+      fullscreenKey = keydownEvent => keycode.isEventKey(keydownEvent, 'f');
+    }
+    if (typeof muteKey !== 'function') {
+      muteKey = keydownEvent => keycode.isEventKey(keydownEvent, 'm');
+    }
+    if (typeof playPauseKey !== 'function') {
+      playPauseKey = keydownEvent => (keycode.isEventKey(keydownEvent, 'k') || keycode.isEventKey(keydownEvent, 'Space'));
+    }
 
-        if (muteToggle && muteToggle.handleClick) {
-          muteToggle.handleClick();
-        } else {
-          this.player_.muted(this.player_.muted() ? false : true);
-        }
-      } else if (keycode.isEventKey(event, 'k')) {
-        // "k" = toggle play/pause
+    // Trigger the ControlBar equivalent for each key, if possible
+    const cb = this.controlBar;
 
-        event.preventDefault();
+    if (fullscreenKey.call(this, event)) {
 
-        const cb = this.player_.controlBar;
-        const playToggle = cb && cb.playToggle;
+      event.preventDefault();
 
-        if (playToggle && playToggle.handleClick) {
-          playToggle.handleClick();
-        }
+      const fullscreenToggle = cb && cb.fullscreenToggle;
+
+      // TODO: make sure that this hotkey doesn't "toggle fullscreen" if
+      //  fullscreen is disabled. For now, ONLY allow fullscreen if the
+      //  fullscreen button is present in the ControlBar.
+
+      if (fullscreenToggle && fullscreenToggle.handleClick) {
```

Comment:
would be nice to make these not depend on the control bar components as they can be disabled.

Reply:
We can do it in a separate PR, as long as it's done before we make a release (I might make a PR against this PR for it)